File: packages/sdk-python/src/agentjail/public.py
"""Public — unauthenticated health + live counters."""

from __future__ import annotations

from ._http import HttpClient
from .types import PublicStats


class Public:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def health(self) -> str:
        """Returns the string ``"ok"`` when the server is up."""
        return self._http.request("GET", "/healthz")

    def stats(self) -> PublicStats:
        return self._http.request("GET", "/v1/stats")

File: packages/sdk-node/src/public.ts
import type { HttpClient } from "./http.js";
import type { PublicStats } from "./types.js";

/** Unauthenticated endpoints — safe to hit without an API key. */
export class Public {
  constructor(private readonly http: HttpClient) {}

  /** Liveness probe. Returns the string `"ok"` when the server is up. */
  async health(): Promise<string> {
    return this.http.request<string>({ method: "GET", path: "/healthz" });
  }

  /** Live counters (active execs, total execs, session count, credential count). */
  async stats(): Promise<PublicStats> {
    return this.http.request<PublicStats>({ method: "GET", path: "/v1/stats" });
  }
}
